google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/strategy/CreateNodesWithGroupEncodedIntoNameThenAddToSet.java

```diff
@@ -134,52 +128,51 @@ protected CreateNodesWithGroupEncodedIntoNameThenAddToSet(
     * Try and find a network either previously created by jclouds or user defined.
     */
    private Network getOrCreateNetwork(GoogleComputeEngineTemplateOptions templateOptions, String sharedResourceName) {
-
       String networkName = templateOptions.getNetworkName().or(sharedResourceName);
-
       return networkMap.apply(new NetworkAndAddressRange(networkName, DEFAULT_INTERNAL_NETWORK_RANGE, null));
    }
 
    /**
-    * Ensures that a firewall exists for every inbound port that the instance requests.
+    * Ensures that a firewall exists with all the inbound ports that the instance requests.
     * <p>
-    * For each port, there must be a firewall with a name following the {@link FirewallTagNamingConvention},
-    * with a target tag also following the {@link FirewallTagNamingConvention}, which opens the requested port
-    * for all sources on both TCP and UDP protocols.
+    * For each group of nodes, there must be a firewall which opens the requested ports for all sources on both TCP and UDP protocols.
     * @see org.jclouds.googlecomputeengine.features.FirewallApi#patch(String, org.jclouds.googlecomputeengine.options.FirewallOptions)
     */
-   private void getOrCreateFirewalls(GoogleComputeEngineTemplateOptions templateOptions, Network network,
-                                     FirewallTagNamingConvention naming) {
-
+   private void getAndUpdateOrCreateFirewalls(GoogleComputeEngineTemplateOptions templateOptions, Network network,
+                                             String sharedResourceName) {
+      String firewallName = templateOptions.getNetworkName().or(sharedResourceName);
```

Comment:
Ping. Seeing jclouds-labs-google was not promoted to core if we want to make these kinds of changes now is the time to do it.